File: Backend/TranscomaAPI/Comun/Entidades/Puerto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    /// <summary>
    /// Entidad que almacena los datos de
    /// los puertos de envio
    /// </summary>
    public class Puerto: Entidad
    {
        private string _codigo;
        private string _nombre;
        private Direccion _direccion;


        public Puerto() { }
        /// <summary>
        /// Constructor de la EntidadPuerto
        /// </summary>
        /// <param name="id">identificacion del puerto</param>
        /// <param name="codigo">Codigo del puerto</param>
        /// <param name="nombre">Nombre de =l puerto</param>
        public Puerto(int id, string codigo, string nombre)
        {
            Id = id;
            _codigo = codigo;
            _nombre = nombre;
        }

        public string Codigo { get => _codigo; set => _codigo = value; }
        public string Nombre { get => _nombre; set => _nombre = value; }
        public Direccion Direccion { get => _direccion; set => _direccion = value; }
    }
}

File: frontend/src/app/models/Puerto.ts
import { EntidadBase } from "./EntidadBase";
import { Direccion } from './Direccion';

export class Puerto extends EntidadBase {

    private _codigo: string ;
    private _nombre: string ;
    private _direccion: Direccion;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$Codigo = data.codigo;
            this.$Nombre = data.nombre;
            if(data.direccion != undefined){
                this.$Direccion = new Direccion(data.direccion);
            }
            else{
                this.$Direccion = null;
            }
           
        }
    }

    /** Getters y setters */
    public get $Codigo(): string {
        return this._codigo;
    }

    public set $Codigo(value: string) {
        this._codigo = value;
    }

    public get $Nombre(): string {
        return this._nombre;
    }

    public set $Nombre(value: string) {
        this._nombre = value;
    }

    public get $Direccion(): Direccion {
        return this._direccion;
    }

    public set $Direccion(value: Direccion) {
        this._direccion = value;
    }


}
